packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -2834,3 +2859,101 @@ typedef SelectableRegionContextMenuBuilder = Widget Function(
   BuildContext context,
   SelectableRegionState selectableRegionState,
 );
+
+/// Signature for the callback that reports when the user changes the selection
+/// under a [SelectionListener].
+///
+/// The [selections] list is ordered according to the order of the
+/// [Selectable]s contained under the [SelectionListener] this callback
+/// is provided to.
+///
+/// [selections] will be an empty list if nothing is selected.
+typedef SelectionListenerSelectionChangedCallback = void Function(List<SelectedContentRange<Object>> selections);
+
+/// A [SelectionContainer] that allows the user to listen to selection changes
+/// for the child subtree it wraps under a [SelectionArea] or [SelectableRegion].
+///
+/// This widget should have an ancestor [SelectionArea] or [SelectableRegion]
+/// to be able to listen to selection changes in this widgets subtree.
+///
+/// This widget does not listen to selection changes of nested [SelectionArea]s
+/// or [SelectableRegion]s in its subtree because those widgets create their own
+/// [SelectionRegistrar]s that do not register to any ancestor registrar.
+///
+/// {@tool dartpad}
+/// This example shows how to color red the active selection
```

Comment:
"how to color red the active selection" => "how to color the active selection red"